src/datanode/src/sql/insert.rs

```diff
@@ -123,6 +139,153 @@ impl SqlHandler {
             region_number: 0,
         }))
     }
+
+    fn build_request_from_batch(
+        stmt: Insert,
+        table: TableRef,
+        batch: RecordBatch,
+        query_ctx: QueryContextRef,
+    ) -> Result<SqlRequest> {
+        let (catalog_name, schema_name, table_name) =
+            table_idents_to_full_name(stmt.table_name(), query_ctx)?;
+
+        let columns: Vec<String> = stmt.columns().iter().map(|c| (*c).clone()).collect();
+        let schema = table.schema();
+        let columns = if columns.is_empty() {
+            schema
+                .column_schemas()
+                .iter()
+                .map(|c| c.name.to_string())
+                .collect()
+        } else {
+            columns
+        };
```

Comment:
To avoid an unnecessary collect:

```suggestion
        let schema = table.schema();
        let columns: Vec<String> = if stmt.columns().is_empty() {
            schema
                .column_schemas()
                .iter()
                .map(|c| c.name.to_string())
                .collect()
        } else {
            stmt.columns().iter().map(|c| (*c).clone()).collect()
        };
```